CRVResponse/src/CrvPhotonGenerator_module.cc

```diff
@@ -201,70 +196,59 @@ namespace mu2e
           double mass = particle.ref().mass();  //MeV/c^2
           double charge = particle.ref().charge(); //in units of elementary charges
 
-          double momentum1 = step.momentum().mag(); //MeV/c
-          double energy1 = sqrt(momentum1*momentum1 + mass*mass); //MeV
-//FIXME: does not take the energy of daughter particles into account
-          double energy2 = energy1 - energyDepositedTotal; //MeV
-          if(energy2<mass) energy2=mass;
-
-          double gamma1 = energy1 / mass;
-          double gamma2 = energy2 / mass;
-          double beta1 = sqrt(1.0-1.0/(gamma1*gamma1));
-          double beta2 = sqrt(1.0-1.0/(gamma2*gamma2));
-          double beta = (beta1+beta2)/2.0;
-          double velocity = beta*CLHEP::c_light;
-          double t2 = t1 + step.stepLength()/velocity;
+          double energy1   = sqrt(step.startMom().mag2() + mass*mass); //MeV
+          double energy2   = sqrt(step.endMom().mag2() + mass*mass);
```

Comment:
here only mag is used.

Reply:
I don't use the direction anywhere, but that may change in the future.